pkg/karmadactl/cmdinit/utils/kubeclient.go

```diff
@@ -9,18 +9,23 @@ import (
 )
 
 // RestConfig  Kubernetes kubeconfig
-func RestConfig(kubeconfigPath string) (*rest.Config, error) {
-	config, err := clientcmd.BuildConfigFromFlags("", kubeconfigPath)
+func RestConfig(context, kubeconfigPath string) (*rest.Config, error) {
+	pathOptions := clientcmd.NewDefaultPathOptions()
+
+	loadingRules := *pathOptions.LoadingRules
+	loadingRules.Precedence = pathOptions.GetLoadingPrecedence()
+	loadingRules.ExplicitPath = kubeconfigPath
+	overrides := &clientcmd.ConfigOverrides{
+		CurrentContext: context,
+	}
+	clientConfig := clientcmd.NewNonInteractiveDeferredLoadingClientConfig(&loadingRules, overrides)
+
+	restConfig, err := clientConfig.ClientConfig()
 	if err != nil {
 		return nil, err
 	}
-	config.QPS = float32(5.000000)
-	config.Burst = 10
-	config.ContentType = "application/json"
-	config.AcceptContentTypes = "application/json"
-	config.UserAgent = rest.DefaultKubernetesUserAgent()
```

Comment:
Will these deletions have any impact?

Reply:
There should no impacts. They all have default value.

https://github.com/karmada-io/karmada/blob/2d47c5b08a27132b2c2e2fa6ef080f54ee157c92/vendor/k8s.io/client-go/rest/config.go#L115-L121

https://github.com/karmada-io/karmada/blob/2d47c5b08a27132b2c2e2fa6ef080f54ee157c92/vendor/k8s.io/client-go/rest/config.go#L284-L294

Reply:
got it.

Reply:
What's the reason why we set them explicitly? @prodanlabs